### PR TITLE
storage incentives: add stamper putter to api

### PR DESCRIPTION
**Pull request by acud**
#873 

changes that were made outside the scope of this PR are due to complaints from deepsource.

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -269,10 +275,19 @@ func equalASCIIFold(s, t string) bool {
...
+func requestPipelineFn(s storage.Storer, r *http.Request) pipelineFunc {
 	mode, encrypt := requestModePut(r), requestEncrypt(r)
 	return func(ctx context.Context, r io.Reader, l int64) (swarm.Address, error) {
 		pipe := builder.NewPipelineBuilder(ctx, s, mode, encrypt, nil)
 		return builder.FeedPipeline(ctx, pipe, r, l)
 	}
 }
+
+func requestStamper(post postage.Service, signer crypto.Signer, batch []byte) (postage.Stamper, error) {
+	i, err := post.GetStampIssuer(batch)
```
Comment (acud) [inline pkg/api/api.go:287]: `post.GetStamper(batch,signer)`

File: pkg/file/pipeline/stamp/stamp.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	next    pipeline.ChainWriter
+}
+
+func NewStampWriter(stamper postage.Stamper, next pipeline.ChainWriter) pipeline.ChainWriter {
+	return &stampWriter{stamper: stamper, next: next}
+}
+
+func (w *stampWriter) ChainWrite(p *pipeline.PipeWriteArgs) (err error) {
+	addr := swarm.NewAddress(p.Ref)
+	p.Stamp, err = w.stamper.Stamp(addr)
```
Comment (acud) [inline pkg/file/pipeline/stamp/stamp.go:24]: this sets the stamp on the pipe write args, but does not implement the change which adds the stamp to the actual chunk which is in turn stored on the localstore. i suggest that this change is done in one PR (change to the chunk interface + persistence in localstore)

**Review by zelig — APPROVED** [review]
i wish we kept the unrelated fixes in separate PRs in future

**Comment by acud** [comment]
@zelig but the CI is failing. I don't know any way around that, particularly that deepsource scans different parts of the codebase randomly on different PRs :man_shrugging:

**Review by paxthemax — APPROVED** [review]
Looks good to me.

**Pull request merged**